core/src/main/java/edu/wpi/grip/core/sources/HttpSource.java

```diff
@@ -87,10 +85,10 @@
       GripServer server,
       ContextStore store,
       @Assisted String path) {
-    super(exceptionWitnessFactory);
+    super(makeId(HttpSource.class), exceptionWitnessFactory);
```

Comment:
Why do you need this duplicate code in every source?
Why not just have the super constructor pull this information?